src/services/EcoBridge/Socket/SocketBridge.ts

```diff
@@ -468,42 +402,214 @@ export class SocketBridge extends EcoBridgeChildBase {
 
     const { toAmount, serviceTime, totalGasFeesInUsd, routeId } = bestRoute
 
+    const formattedToAmount = Number(formatUnits(toAmount, result.toAsset.decimals)).toFixed(2).toString()
+
     this.store.dispatch(commonActions.setActiveRouteId(routeId))
 
     const details = {
       gas: `${totalGasFeesInUsd.toFixed(2).toString()}$`,
       estimateTime: `${(serviceTime / 60).toFixed(0).toString()} min`,
-      receiveAmount: Number(formatUnits(toAmount, toAsset.decimals)).toFixed(2).toString(),
+      receiveAmount: formattedToAmount,
       requestId: helperRequestId,
     }
 
     this.store.dispatch(this.actions.setBridgeDetails(details))
   }
 
+  public getBridgingMetadata = async () => {
+    const isBridgeSwapActive = this.store.getState().ecoBridge.ui.isBridgeSwapActive
+    const requestId = this.store.getState().ecoBridge[this.bridgeId as SocketList].lastMetadataCt
+
+    const helperRequestId = (requestId ?? 0) + 1
+
+    this.store.dispatch(this.actions.requestStarted({ id: helperRequestId }))
+
+    // reset previous data
+    this.store.dispatch(
+      this.actions.setApprovalData({
+        allowanceTarget: undefined,
+        amount: undefined,
+        chainId: undefined,
+        owner: undefined,
+        tokenAddress: undefined,
+      })
+    )
+    this.store.dispatch(this.actions.setRoutes([]))
+
+    this.store.dispatch(this.actions.setBridgeDetailsStatus({ status: SyncState.LOADING }))
+
+    const { from, to } = this.store.getState().ecoBridge.ui
+
+    if (!from.chainId || !to.chainId || !this._account || !from.address || Number(from.value) === 0) return
+
+    let value = BigNumber.from(0)
+
+    //handling small amounts
+    try {
+      value = parseUnits(from.value, from.decimals)
+    } catch (e) {
+      this.store.dispatch(
+        this.actions.setBridgeDetailsStatus({
+          status: SyncState.FAILED,
+          errorMessage: 'No available routes / details',
+        })
+      )
+      return
+    }
+
+    if (isBridgeSwapActive) {
+      await this._fetchSocketRoutes({
+        fromChainId: from.chainId.toString(),
+        fromTokenAddress:
+          from.address === Currency.getNative(from.chainId).symbol ? SOCKET_NATIVE_TOKEN_ADDRESS : from.address,
```

Comment:
Symbol is risky :) 

Hint:
`SOCKET_NATIVE_TOKEN_ADDRESS === Currency.getNative(from.chainId).address` is true :)

Reply:
I will try to explain why we are comparing `from.address` to `Currency.getNative(from.chainId).symbol`.
When user select native currency we are doing dispatch to `store`.
```js
{
  chainId:1
  address:"ETH"
  decimals:18
  name:"Ether"
  symbol:"ETH"
}
```
As you can see the `address` is actually `symbol` - (this is only for native currencies)
It works like that because inside dispatch we are using `currencyId` function.
```js
dispatch(
  ecoBridgeUIActions.setFrom({
    address: currency instanceof Currency ? currencyId(currency) : currency,
    //....
  })
)        
```
And `currencyId` function looks like:
```js
export function currencyId(currency: Currency): string {
  if (Currency.isNative(currency)) return currency.symbol || ''
  if (currency instanceof Token) return currency.address
  throw new Error('invalid currency')
}
```
As you can see when we have native currency function returns symbol and that's why we have symbol under address property

Reply:
Ok, that clears it up. I suggest we, revisit this implementation and make sure the variable names describe their content :)